shell.go

```diff
@@ -0,0 +1,256 @@
+package devbox
```

Comment:
I didn't put it in another package because I didn't want to split things up too much, especially if we want devbox to be used as a library. `devbox.PackageStore` reads nicely and `nixless.PackageStore` isn't really true.

I'd prefer to keep it in `devbox` for now, but also happy to discuss more if you'd like.

Reply:
No worries. You can leave as is, and we can always refactor later if the devbox package starts to grow too much.